webui/react/src/pages/WorkspaceDetails/WorkspaceQuickSearch.tsx

```diff
@@ -0,0 +1,154 @@
+import { Input, Modal, Tree } from 'antd';
+import type { DefaultOptionType } from 'rc-tree-select/lib/TreeSelect';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+
+import Link from 'components/Link';
+import { paths } from 'routes/utils';
+import { getWorkspaceProjects, getWorkspaces } from 'services/api';
+import Icon from 'shared/components/Icon/Icon';
+import Message, { MessageType } from 'shared/components/Message';
+import Spinner from 'shared/components/Spinner';
+import { ErrorLevel, ErrorType } from 'shared/utils/error';
+import { routeToReactUrl } from 'shared/utils/routes';
+import { Project, Workspace } from 'types';
+import handleError from 'utils/error';
+
+import css from './WorkspaceQuickSearch.module.scss';
+
+interface Props {
+  children: React.ReactChild;
+}
+
+const WorkspaceQuickSearch: React.FC<Props> = ({ children }: Props) => {
+  const [ searchText, setSearchText ] = useState<string>('');
+  const [ workspaceMap, setWorkspaceMap ] = useState<Map<Workspace, Project[]>>(new Map());
+  const [ isLoading, setIsLoading ] = useState(true);
+  const [ isModalVisible, setIsModalVisible ] = useState(false);
+  const canceler = useRef(new AbortController());
+
+  const fetchData = useCallback(async () => {
+    try {
+      const workspaceResponse = await getWorkspaces(
+        { limit: 0, sortBy: 'SORT_BY_NAME' },
+        { signal: canceler.current.signal },
+      );
+      const filteredWorkspaces = workspaceResponse.workspaces.filter((w) => !w.immutable);
+      const projectAPIList = filteredWorkspaces
+        .map((workspace) =>
+          getWorkspaceProjects(
+            { id: workspace.id, sortBy: 'SORT_BY_NAME' },
+            { signal: canceler.current.signal },
+          ));
+      const projectResponse = (await Promise.all(projectAPIList))
+        .map((project) => project.projects);
+
+      // Promise.all preserves the order
+      const tempWorkspaceMap: Map<Workspace, Project[]> = new Map();
+      for (const workspace of filteredWorkspaces.reverse()) {
+        const projects = projectResponse.pop();
+        tempWorkspaceMap.set(workspace, projects ?? []);
+      }
+      setWorkspaceMap(tempWorkspaceMap);
+    } catch (e) {
+      handleError(e, {
+        level: ErrorLevel.Error,
+        publicMessage: 'Please try again later.',
+        publicSubject: 'Unable to fetch data.',
+        silent: false,
+        type: ErrorType.Api,
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  }, [ ]);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchText(e.target.value.toLocaleLowerCase());
```

Comment:
this changes as user is typing, no? 

probably can just keep the lowercasing to where you're doing the comparison

Reply:
it doesnt cuz i didnt add value property but its kinda vague so i'll change it